src/form-elements/FormElementCompliance.tsx

```diff
@@ -92,44 +81,89 @@ function FormElementCompliance({
     element,
   ])
 
-  const handleValueChange = React.useCallback(
-    (fe: FormTypes.FormElement, v: unknown) => {
-      onChange(fe, {
-        ...typedValue,
-        value: v,
+  const handleValueChange = React.useCallback<
+    FormElementValueChangeHandler<string>
+  >(
+    (fe, v) => {
+      onChange(fe, (existingValue: Value | undefined) => {
+        let newValue = undefined
+        if (typeof v === 'function') {
+          newValue = v(existingValue ? existingValue.value : undefined)
+        } else {
+          newValue = v
+        }
+        if (!newValue) {
+          return
+        }
+        return {
+          ...existingValue,
+          value: newValue,
+        }
       })
     },
-    [onChange, typedValue],
+    [onChange],
   )
-  const handleNotesChange = React.useCallback(
-    (v: unknown) => {
-      const newNotes = !!v && typeof v === 'string' ? v : undefined
-      onChange(element, { ...typedValue, notes: newNotes })
+  const handleNotesChange = React.useCallback<
+    React.ComponentProps<typeof FormElementTextarea>['onChange']
+  >(
+    (fe, v) => {
+      onChange(element, (existingValue) => {
+        if (!existingValue) {
+          return
+        }
+        let newNotes = undefined
+        if (typeof v === 'function') {
+          newNotes = v(existingValue.notes)
+        } else {
+          newNotes = v
+        }
+        return {
+          ...existingValue,
+          notes: newNotes,
+        }
+      })
     },
-    [element, onChange, typedValue],
+    [element, onChange],
   )
-  const handleFilesChange = React.useCallback(
-    (v: PossibleFileConfiguration[] | undefined) => {
-      const newFiles = v ? v : undefined
-      onChange(element, { ...typedValue, files: newFiles })
+  const handleFilesChange = React.useCallback<
+    React.ComponentProps<typeof FormElementFiles>['onChange']
+  >(
+    (fe, v) => {
```

Comment:
Why do we need `fe` here? ie. why pass the element to the hook, only to pass it back here and seemingly do nothing with it?

Reply:
`<FormElementFiles />` `onChange()` passes back the form element. If we leave it the way it is, the onChange prop will change every time this component renders.